src/resolver.ts

```diff
@@ -71,6 +71,8 @@ export async function resolveDeno(
   id: string,
   cwd: string,
 ): Promise<DenoResolveResult | null> {
+  if (id.startsWith("\x00")) return null; // ignore vite virtual modules
```

Comment:
I tested this locally, but added a bit more just to ensure it was only targetting virtual modules.

```typescript 
if (id.startsWith("\x00virtual:"))
        return null;
```

It's up to you if you want to target the more specific syntax or not, but either way it worked properly for me and HMR started working again for me locally.